include/uapi/sound/sof/header.h

```diff
@@ -26,6 +26,20 @@ struct sof_abi_hdr {
 	__u32 data[0];		/**< Component data - opaque to core */
 }  __packed;
 
+/*
+ * Identifies data type, size and ABI for IPC4.
+ * Used by any bespoke component data structures or binary blobs.
+ */
+struct sof_ipc4_abi_hdr {
+	__u32 magic;		/**< 'S', 'O', 'F', '4' */
+	__u32 size;		/**< size in bytes of data excl. this struct */
+	__u32 abi;		/**< SOF ABI version */
+	__u32 ipc_blob_type;	/** IPC4 type. 0: init, 2: config, 4: large_config */
+	__u32 ipc_extension;	/** used in IPC4 module config/large_config */
```

Comment:
Note that module config_set/config_get is used to exchange small piece of data between linux and firmware, the data is encoded in ipc_extension.  do we want to use `ipc_extension` or `data` in sof_ipc4_abi_hdr?, since you comment `2:config` for ipc_blob_type.

Reply:
If there is no ipc_extension or data, just set them to 0. We will support several types of kcontrol, so we need a full support for the field. For example, for large config, we need both ipc_extension and data, then we will set them both. For set_config as you mentioned, we can just set extension and leave data as 0 and set the size to 0. The driver will automatically ignore the data.

Reply:
Yeah, I know, I mean you should tell people that on config, they should put data in `ipc_extension`, instead of `data`. you should add comments for this struct, how it is used under each condition.

Reply:
@aiChaoSONG OK. I suppose the tplg developer should know this knowledge before he develop the tplg

Reply:
suppose nothing. I bet very few people in our team know this, let along the community

Reply:
> @aiChaoSONG OK. I suppose the tplg developer should know this knowledge before he develop the tplg

Please don't assume people know anything. :)

Reply:
completely agree, knowledge about such details of IPC4 are not known widely and need to be documented with comments.